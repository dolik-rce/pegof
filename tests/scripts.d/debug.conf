input scripts.d/scripts.peg
header never
optimize inline,remove-group
debug-script scripts.d/debug.sh
