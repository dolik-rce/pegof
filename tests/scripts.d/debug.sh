#!/usr/bin/env bash

set -e
echo "Debug script called: $*"
# Let's fail when rule C is inlined
if ! grep -q "C" "$1"; then
    echo "Detected problem in $1:"
    cat "$1"
    exit 42
fi
