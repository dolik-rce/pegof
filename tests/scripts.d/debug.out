Running debug script scripts.d/debug.sh...
Debug script called: /tmp/pegof_<random>/pass_1.peg
Running debug script scripts.d/debug.sh...
Debug script called: /tmp/pegof_<random>/pass_2.peg
Running debug script scripts.d/debug.sh...
Debug script called: /tmp/pegof_<random>/pass_3.peg
Detected problem in /tmp/pegof_<random>/pass_3.peg:
X <- A

A <- B

B <- "x"
ERROR: Debug script failed in pass 3! (exit_code=42)
